Yoast/wordpress-seo
# 12661 implement feature flag functionality

## Summary

This PR can be summarized in the following changelog entry:

* [not-user-facing] Adds general feature toggle functionality to the plugin for toggling experimental features on (or off).

## Relevant technical choices:

* Enables feature flags in javascript in three places:
  * Inside of `wp-seo-metabox.js`, for javascript functionality outside of the web worker.
  * Inside of the web worker, through a configuration option on the `initialize` message to the web worker. This makes the feature flags available to the web workers, since it is run in a separate thread. 
  * Inside of `site-wide-analysis` in premium (see https://github.com/Yoast/wordpress-seo-premium/pull/2340).

## Test instructions

This PR can be tested by following these steps:

* Checkout this branch (`12661-implement-feature-flag-functionality`) on your local `wordpress-seo` repository.
* Checkout `12661-implement-feature-flag-functionality` on your local `javascript` repository.
  * **Tip**: you can use the `yarn link-monorepo` command inside of the `wordpress-seo` folder to automatically checkout the right branch and link the right packages.
* Enable features by adding this line of code to the `wp-config.php` file (inside of the root folder of your WordPress installation (e.g. `.../vvv/www/wordpress-default/public_html/wp-config.php`)):
  ```php
  define( 'YOAST_SEO_ENABLED_FEATURES', 'some-feature, another-feature' );
  ```
* Anywhere inside of the web worker, like in a Researcher or Assessment, add this line to make sure that the feature flags are coming through in the analysis:
  * `console.log( "Currently enabled features: ", enabledFeatures() )`.
* Start your local `wordpress-seo` environment in development mode.
* Open a post.
* You should see a line in the console saying:
  * `Currently enabled features: ["some-feature", "another-feature"]`.

## UI changes

## Quality assurance


Fixes #12661
